python/tests/unit/arcticdb/version_store/test_read_index.py

```diff
@@ -0,0 +1,413 @@
+        lib = ac.create_library(lib_name, LibraryOptions(dynamic_schema=dynamic_schema, rows_per_segment=5, columns_per_segment=2))
+        row_range = (3, 8)
+        lib.write("sym", pd.DataFrame({"col": range(0, len(index))}, index=index))
+        result = lib.read("sym", row_range=row_range, columns=[])
+        assert isinstance(result, arcticdb.VersionedItem)
+        assert result.symbol == "sym"
+        assert result.version == 0
+        assert result.data.index.equals(index[row_range[0]:row_range[1]])
+        assert result.data.empty
+
+    @pytest.mark.parametrize("non_datetime_index", [
+        pd.RangeIndex(start=0, stop=5),
+        pd.MultiIndex.from_arrays(
+            [pd.RangeIndex(start=10, stop=21), pd.date_range(start="01/11/2024", end="01/21/2024")],
+            names=["range", "date"]
+        )
+    ])
+    @pytest.mark.parametrize("dynamic_schema", [False, True])
+    def test_date_range_throws(self, lmdb_storage, lib_name, dynamic_schema, non_datetime_index):
+        ac = lmdb_storage.create_arctic()
+        lib = ac.create_library(lib_name, LibraryOptions(dynamic_schema=dynamic_schema))
+        lib.write("sym", pd.DataFrame({"col": list(range(0, len(non_datetime_index)))}, index=non_datetime_index))
+        with pytest.raises(Exception):
+            lib.read("sym", date_range=(datetime(2024,1,4), datetime(2024,1,10)), columns=[])
+
+
+class TestWithNormalizers:
+
+    @pytest.mark.parametrize("dynamic_schema", [False, True])
+    def test_recursive_throws(self, lmdb_storage, lib_name, dynamic_schema):
+        data = {"a": np.arange(5), "b": np.arange(8)}
+        ac = lmdb_storage.create_arctic()
+        lib = ac.create_library(lib_name, LibraryOptions(dynamic_schema=dynamic_schema))
+        lib._nvs.write("sym_recursive", data, recursive_normalizers=True)
+        with pytest.raises(InternalException) as exception_info:
+            lib.read("sym_recursive", columns=[])
+        assert "Reading the index column is not supported when recursive or custom normalizers are used." in str(exception_info.value)
+
+    @pytest.mark.parametrize("dynamic_schema", [False, True])
+    def test_custom_throws(self, lmdb_storage, lib_name, dynamic_schema):
+        register_normalizer(TestCustomNormalizer())
+        ac = lmdb_storage.create_arctic()
+        lib = ac.create_library(lib_name, LibraryOptions(dynamic_schema=dynamic_schema))
+        data = CustomThing(custom_columns=["a", "b"], custom_index=[12, 13], custom_values=[[2.0, 4.0], [3.0, 5.0]])
+        lib._nvs.write("sym_custom", data)
+
+        with pytest.raises(InternalException) as exception_info:
+            lib.read("sym_custom", columns=[])
+        assert "Reading the index column is not supported when recursive or custom normalizers are used." in str(exception_info.value)
+
+
+class TestReadBatch:
+    @pytest.mark.parametrize("dynamic_schema", [False, True])
+    def test_read_batch(self, lmdb_storage, lib_name, dynamic_schema, index):
```

Comment:
Need a test with batch read and row-range arguments

Reply:
batch read's row_range is generally broken: #1692